### PR TITLE
DOC8 - Configuration for documentation check

### PR DESCRIPTION
**Pull request by Prome88**
DOC8 - Documentation fixes for - 002 Trailing Whitespace

### CODE REVIEW

**Comment by atodorov** [comment]
@Prome88 bring your suggestions from #1000 into this PR so we can review and comment on them. GH UI is not suited for commenting on big code clocks as issue comments.

File: doc8.ini [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[doc8]
+path=/docs/source
```
Comment (atodorov) [inline doc8.ini]: Missing new line at end of file

Comment (atodorov) [inline doc8.ini]: Should this be `docs/source` instead ? (no leading `/`) ? Or `./docs/source` (which is the same) .

Travis CI shows errors and you can see it also inspects CHANGELOG.rst and some other files which are not in this directory.

Comment (Prome88) [reply doc8.ini]: It's the correct syntax. CHANGELOG.rst is referenced in /docs/changelog.rst and therefore checked by doc8. I'll add the file to ignored list.

Comment (atodorov) [reply doc8.ini]: It is OK if we are referencing this file, don't ignore it. However what about these:

```
/home/travis/build/kiwitcms/Kiwi/tcms/templates/email/post_run_save/email.txt
/home/travis/build/kiwitcms/Kiwi/tcms/node_modules/datatables.net-bs/License.txt
```

it looks like doc8 just finds all txt files and checks them, which is not what we want.

Comment (Prome88) [reply doc8.ini]: Added extension parameter and excluded txt folders which have referenced files, as doc8 by default checks for .rst and .txt files.

File: .travis.yml [hunk 2]
```diff
@@ -112,6 +113,10 @@ jobs:
           packages:
             - graphviz
 
+    - stage: non_functional_testing
+      env:
+        - CMD=doc8
+
```
Comment (atodorov) [inline .travis.yml]: remove extra white space here

File: Makefile [hunk 3]
```diff
@@ -98,6 +98,10 @@ check-docs-source-in-git: docs
 	    exit 1; \
 	fi
 
+.PHONY: doc8
+doc8:
+	doc8 --config="doc8.ini"
```
Comment (atodorov) [inline Makefile]: --config is redundant b/c doc8.ini in the current directory is the default name for the config file

Comment (atodorov) [reply Makefile]: The path(s) to examine are given as positional parameter to the doc8 command, so `doc8 docs/source/`

File: doc8.ini [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+extensions=.rst
+ignore-path=tcms/templates/*,requirements/*
```
Comment (atodorov) [inline doc8.ini]: This file is useless as-is so remove it:

- `path` is not a config file option
-  `extensions` is not a config file option
- `ignore-path` - still doesn't ignore files ourside docs/sources or txt files

**Comment by atodorov** [comment]
Also when ready:
- squash all commits together
- rebase to latest master
- add 'Fixes #1000' at the end of the 1st line in commit log.

File: doc8.ini (hunk 4 above)
Comment (Prome88) [reply doc8.ini]: - ignore-path enabled - 130 files scanned, disabled 140 files scanned
- extensions should be extension
- path will be removed and scan location invoked via cmd

Comment (atodorov) [reply doc8.ini]: This file is not needed when we run `make doc8` so can you explain why do we need it ?

**Comment by atodorov** [comment]
Next time don't close the pull request, just rebase (instead of merge) and force push and it will be fine. If you have the branch locally you can try reopening and pushing the branch again, not sure if it will help.

**Pull request closed without merging**